pkg/ccl/backupccl/restore_old_versions_test.go

```diff
@@ -730,3 +744,44 @@ func TestRestoreWithDroppedSchemaCorruption(t *testing.T) {
 	}
 	require.Falsef(t, hasSameNameSchema(dbName), "corrupted descriptor exists")
 }
+
+// restorePublicSchemaRemap tests that if we're restoring a database from
+// an older version where the database has a synthetic public schema, a real
+// descriptor backed public schema is created and the tables in the schema
+// are correctly mapped to the new public schema.
+func restorePublicSchemaRemap(exportDir string) func(t *testing.T) {
+	return func(t *testing.T) {
+		const numAccounts = 1000
+		_, _, _, tmpDir, cleanupFn := BackupRestoreTestSetup(t, MultiNode, numAccounts, InitManualReplication)
+		defer cleanupFn()
+
+		_, _, sqlDB, cleanup := backupRestoreTestSetupEmpty(t, singleNode, tmpDir,
+			InitManualReplication, base.TestClusterArgs{})
+		defer cleanup()
+		err := os.Symlink(exportDir, filepath.Join(tmpDir, "foo"))
+		require.NoError(t, err)
+
+		sqlDB.Exec(t, fmt.Sprintf("RESTORE DATABASE %s FROM '%s'", "d", LocalFoo))
```

Comment:
Maybe just RESTORE DATABASE d instead of the string formatter %s?

Reply:
Done